raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2216,7 +2316,7 @@ public void complete() {
         // These fields are visible to both the Raft IO thread and the listener
         // and are protected through synchronization on this `ListenerContext` instance
         private BatchReader<T> lastSent = null;
-        private long lastAckedOffset = 0;
+        private long lastAckedEndOffset = 0;
```

Comment:
I am always confused if an given offset is inclusive or exclusive. I was thinking that by adding "end offset" it would make it clear that the offset is exclusive. I can write a comment instead if you prefer.

Reply:
I don't feel too strongly about it, but the new name is a little confusing to me. Why would the client only be acking end offsets? Especially confusing when I see this `lastAckedEndOffset = logStartOffset` 🙂 .  I think we probably need a comment here regardless.

Reply:
How about just calling it `nextOffset`?